File: crates/laya-parse/tests/fixtures/refs_sample.ts
import { readFile } from "fs";
import Store, { Entry as E, openStore } from "./store";

export class Cache extends BaseCache implements Lookup {
  private store: Store;

  lookup(key: string): Entry | undefined {
    const raw = this.store.fetchRaw(key);
    console.log(raw);
    return decode<Entry>(raw);
  }
}

export function build(): Cache {
  return new Cache(createStore());
}

File: crates/laya-parse/tests/fixtures/refs_sample.go
package cache

import (
	"fmt"
	"github.com/acme/moon/store"
)

type Cache struct {
	inner *store.Store
	meta  Metadata
}

func (c *Cache) Lookup(key string) (Entry, error) {
	raw, err := c.inner.FetchRaw(key)
	if err != nil {
		return Entry{}, fmt.Errorf("lookup: %w", err)
	}
	return decode(raw), nil
}
